modules/agar/src/main/ts/ephox/agar/api/UiControls.ts

```diff
@@ -6,13 +7,30 @@ import * as UiFinder from './UiFinder';
 
 type TogglableElement = HTMLInputElement | HTMLTextAreaElement | HTMLSelectElement | HTMLOptionElement | HTMLButtonElement;
 
-const setValue = (element: SugarElement<TogglableElement>, newValue: string): void => {
+const fireEvent = (elem: SugarElement, event: string) => {
+  let evt: Event;
+  if (Type.isFunction(Event)) {
+    evt = new Event(event, {
+      bubbles: true,
+      cancelable: true
+    });
+  } else { // support IE
+    evt = document.createEvent('Event');
+    evt.initEvent(event, true, true);
+  }
+  elem.dom.dispatchEvent(evt);
+};
```

Comment:
Maybe like this to avoid let and `true`, `true` in the initEvent?!
```
const fireEvent = (elem: SugarElement, event: string) => {
  const bubbles = true;
  const cancelable = true;
  if (Type.isFunction(Event)) {
    const newEvent = new Event(event, { bubbles, cancelable });
    elem.dom.dispatchEvent(newEvent);
  } else { // support IE
    const newEvent = document.createEvent('Event');
    newEvent.initEvent(event, bubles, cancelable);
    elem.dom.dispatchEvent(newEvent);
  }
};
```

Reply:
```suggestion
const fireEvent = (elem: SugarElement, eventName: string) => {
  const bubbles = true;
  const cancelable = true;
  if (Type.isFunction(Event)) {
    const event = new Event(eventName, { bubbles, cancelable });
    elem.dom.dispatchEvent(event);
  } else { // support IE
    const event = document.createEvent('Event');
    event.initEvent(eventName, bubbles, cancelable);
    elem.dom.dispatchEvent(event);
  }
};
```

Reply:
I don't mind this, anyone else have thoughts. FWIW I just took the implementation that we have plastered all over the place so I didn't really think much about it

Reply:
@diogo405 I might just link to this in the JIRA for doing the rest of this so it doesn't hold up this PR

Reply:
I don't like the duplication with the `elem.dom.dispatchEvent(event);` part. The cancelable and bubbles part is nice, but also becomes obsolete as soon as we drop IE 11 :shrug: